packages/app-mobile/src/containers/Toolbar.tsx

```diff
@@ -53,44 +57,54 @@ const Toolbar: React.FC<ToolbarProps> = (props) => {
     return undefined;
   }, [state.interactionState]);
 
-  const isButtonActive = (shape: DrawableLayerType) =>
-    layerType === shape && (interType === 'drawing' || interType === 'insert');
+  const isButtonActive = useCallback(
+    (shape: DrawableLayerType) =>
+      layerType === shape &&
+      (interType === 'drawing' || interType === 'insert'),
+    [layerType, interType],
+  );
 
-  const onReset = () => {
+  const onReset = useCallback(() => {
     if (interType !== 'none') {
       dispatch('interaction', ['reset']);
     }
-  };
+  }, [dispatch, interType]);
 
-  const onAddShape = (shape: DrawableLayerType) => () => {
-    dispatch('interaction', ['insert', shape]);
-  };
-
-  const processImage = (base64Image: string, uri: string) => {
-    const data = base64ToArrayBuffer(base64Image);
-    const decodedImage = CanvasKit.MakeImageFromEncoded(data);
-    const extension =
-      uri.substring(uri.lastIndexOf('.')) === 'jpg' ? 'jpg' : 'png';
-    const name = uri.substring(uri.lastIndexOf('/'), uri.lastIndexOf('.'));
+  const onAddShape = useCallback(
+    (shape: DrawableLayerType) => () => {
+      dispatch('interaction', ['insert', shape]);
+    },
+    [dispatch],
+  );
 
-    if (!decodedImage) {
-      return;
-    }
+  const processImage = useCallback(
+    (base64Image: string, uri: string) => {
+      const data = base64ToArrayBuffer(base64Image);
+      const decodedImage = CanvasKit.MakeImageFromEncoded(data);
+      const extension =
+        uri.substring(uri.lastIndexOf('.')) === 'jpg' ? 'jpg' : 'png';
+      const name = uri.substring(uri.lastIndexOf('/'), uri.lastIndexOf('.'));
```

Comment:
There's a noya util `delimitedPath` that might simplify this string manipulation